### PR TITLE
chore(html): add HTML class

### PR DESCRIPTION
**Pull request by MrCloudSec**
### Description

Add HTML class that is needed to modularize the code for the new outputs.

### License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: prowler/lib/outputs/html/html.py
Comment (pedrooot) [inline prowler/lib/outputs/html/html.py:534]: Could you add tests for this method for each provider?

**Review by pedrooot — COMMENTED** [review]
Loving this! Please review my comments 😄

File: prowler/__main__.py
Comment (jfagoagas) [inline prowler/__main__.py:354]: Do you need the whole provider object here?

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]: You shouldn't need to do this. Please review it.
```suggestion

```

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]:
```suggestion

```

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]:
```suggestion
```

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]:
```suggestion
```

File: tests/lib/outputs/html/html_test.py
Comment (jfagoagas) [inline tests/lib/outputs/html/html_test.py]: Is this needed?

**Review by jfagoagas — CHANGES_REQUESTED** [review]
That's a great improvement @sergargar, please check my comments and some tests that are failing.

File: tests/lib/outputs/html/html_test.py
Comment (pedrooot) [inline tests/lib/outputs/html/html_test.py]:
```suggestion
        # Clear the data from HTML class
```
Make this for all the comments please 🙏🏼

File: prowler/__main__.py
Comment (MrCloudSec) [reply prowler/__main__.py:354]: Yes, I finally need the provider object since the header need to access to different attributes for each kind of provider when getting the assessments.

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Done

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Done

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Done

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Done

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Nope, removed!

File: tests/lib/outputs/html/html_test.py
Comment (MrCloudSec) [reply tests/lib/outputs/html/html_test.py]: Done

File: tests/lib/outputs/html/html_test.py
Comment (pedrooot) [inline tests/lib/outputs/html/html_test.py]: from tests.lib.outputs.fixtures.fixtures you could use generate_finding_output, wait until #4368 it's merged

**Review by pedrooot — COMMENTED** [review]
We are close!!

File: tests/lib/outputs/html/html_test.py
Comment (pedrooot) [reply tests/lib/outputs/html/html_test.py]: This PR have several changes for this method and we could remove this fixture

File: prowler/lib/outputs/html/html.py
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py:32]: Please add a test for the muted.

File: prowler/lib/outputs/html/html.py
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py:33]: Please add a test for the manual.

File: prowler/lib/outputs/html/html.py
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py:1]: Could you remove all the `sys.exit()` from the code? We cannot abort the execution but raise an exception.

**Review by jfagoagas — CHANGES_REQUESTED** [review]

File: prowler/lib/outputs/html/html.py
Comment (jfagoagas) [inline prowler/lib/outputs/html/html.py]: Why did you remove the spacing?

File: prowler/lib/outputs/html/html.py
Comment (MrCloudSec) [reply prowler/lib/outputs/html/html.py:1]: Done!

File: prowler/lib/outputs/html/html.py
Comment (MrCloudSec) [reply prowler/lib/outputs/html/html.py]: I add it again!

File: prowler/lib/outputs/html/html.py
Comment (jfagoagas) [reply prowler/lib/outputs/html/html.py:1]: Please change the critical to error and add a return value for the function

Comment (MrCloudSec) [reply prowler/lib/outputs/html/html.py:1]: Done! I've added an empty string as a return for the functions that don't have a None return value.

**Review by jfagoagas — DISMISSED** [review]
Well done @sergargar 👏

**Review by jfagoagas — APPROVED** [review]

**Pull request merged**
